ourhexenv.py

```diff
@@ -154,8 +154,9 @@ def reset(self):
         self.board = np.zeros((self.board_size, self.board_size), dtype=int)
         
         self.is_first = True
-        self.is_pie_rule_usable = True
```

Comment:
Why are we removing this? Once the board is reset we need to reset this variable too. Need to initialize the flags on init and reset, probably smthn like this: 
```
self.pie_rule_used = False
self.is_pie_rule_usable = True
```